t/23_utils_git.t

```diff
@@ -0,0 +1,38 @@
+use strict;
+use warnings;
+use testapi;
+use Test::MockModule;
+use Test::Exception;
+use Test::More;
+
+use Utils::Git;
+
+subtest '[git_clone] Compose command' => sub {
+    my $mock_git = Test::MockModule->new('Utils::Git', no_auto => 1);
+    my @calls;
+    $mock_git->redefine(assert_script_run => sub { @calls = $_[0]; return 0; });
+    $mock_git->redefine(record_info => sub { return 0; });
+    $mock_git->redefine(upload_logs => sub { return 0; });
+    my $repository = 'https://github.com/os-autoinst/os-autoinst-distri-opensuse';
+
+    git_clone($repository);
+    is join(' ', @calls), "git clone $repository", "Check base command composition";
+
+    git_clone($repository, skip_ssl_verification => 'true', output_log_file => 'output.log', branch => 'world_domination');
+    note("\n  -->  " . join("\n  -->  ", @calls));
+    ok(grep(/git.*-c http.sslVerify=false.*clone/, @calls), 'Option "-c http.sslVerify=false " must be between "git" and "clone"');
```

Comment:
Does grep works with @calls that is a list? usually we use a different pattern like:

ok((any { /git.*clone.*github.*com\/SUSE\/qe-sap-deployment.*\/BRUCE/ } @calls), 'Default repo cloned in /BRUCE');

Reply:
Yes, I just checked if it fails with grepping something weird. 
Grep works primarily with lists. 
https://perldoc.perl.org/functions/grep